components/match2/wikis/warcraft/match_group_input_custom.lua

```diff
@@ -0,0 +1,956 @@
+
+	--determine score, resulttype, walkover and winner
+	match['map' .. i] = WarcraftMatchGroupInput.MapWinnerProcessing(match['map' .. i])
+
+	--get participants data for the map + get map mode + winnerrace and loserrace
+	--(w/l race stuff only for 1v1 maps)
+	match['map' .. i] = WarcraftMatchGroupInput.ProcessPlayerMapData(match['map' .. i], match, 2)
+
+	--set sumscore to 0 if it isn't a number
+	if match.opponent1.sumscore == '' then
+		match.opponent1.sumscore = 0
+	end
+	if match.opponent2.sumscore == '' then
+		match.opponent2.sumscore = 0
+	end
+
+	--adjust sumscore for winner opponent
+	if (tonumber(match['map' .. i].winner or 0) or 0) > 0 then
+		match['opponent' .. match['map' .. i].winner].sumscore =
+			match['opponent' .. match['map' .. i].winner].sumscore + 1
+	end
+
+	--handle subgroup stuff if team match
+	if string.find(match.mode, 'team') then
+		match['map' .. i].subgroup = tonumber(match['map' .. i].subgroup or '')
+		if match['map' .. i].subgroup then
+			subgroup = match['map' .. i].subgroup
+		else
+			subgroup = subgroup + 1
+			match['map' .. i].subgroup = subgroup
+		end
+	end
+
+	return match, subgroup
+end
+
+function WarcraftMatchGroupInput.MapWinnerProcessing(map)
+	map.scores = {}
+	local manual_scores = false
+	local indexedScores = {}
+	for scoreIndex = 1, MAX_NUM_OPPONENTS do
+		-- read scores
+		local score = map['score' .. scoreIndex]
+		local obj = {}
+		if not Logic.isEmpty(score) then
+			manual_scores = true
+			score = ALLOWED_STATUSES2[score] or score
+			if Logic.isNumeric(score) then
+				obj.status = 'S'
+				obj.score = score
+			elseif Table.includes(ALLOWED_STATUSES, score) then
+				obj.status = score
+				obj.score = -1
+			end
+			table.insert(map.scores, score)
+			indexedScores[scoreIndex] = obj
+		else
+			break
+		end
+	end
+
+	if manual_scores then
+		for scoreIndex, _ in Table.iter.spairs(indexedScores, WarcraftMatchGroupInput.placementSortFunction) do
+			if not tonumber(map.winner or '') then
+				map.winner = scoreIndex
+			else
+				break
+			end
+		end
+	else
+		if (map.walkover or '') ~= '' then
+			if map.walkover == '1' then
+				map.winner = '1'
+			elseif map.walkover == '2' then
+				map.winner = '2'
+			elseif map.walkover == '0' then
+				map.winner = '0'
+			end
+			map.walkover = Table.includes(ALLOWED_STATUSES, map.walkover) and map.walkover or 'L'
+			map.scores = { -1, -1 }
+			map.resulttype = 'default'
+		elseif map.winner == 'skip' then
+			map.scores = { 0, 0 }
+			map.scores = { -1, -1 }
+			map.resulttype = 'np'
+		elseif map.winner == '1' then
+			map.scores = { 1, 0 }
+		elseif map.winner == '2' then
+			map.scores = { 0, 1 }
+		elseif map.winner == '0' or map.winner == 'draw' then
+			map.scores = { 0.5, 0.5 }
+			map.resulttype = 'draw'
+		end
+	end
+	return map
+end
+
+function WarcraftMatchGroupInput.ProcessPlayerMapData(map, match, OppNumber)
+	local participants = {}
+	local map_mode = ''
+	local raceOP = {}
+	local PL = {}
```

Comment:
what is that supposed to be?
variables should use camelCase

Reply:
Copied from starcraft
https://github.com/Liquipedia/Lua-Modules/blob/main/components/match2/commons/starcraft_starcraft2/match_group_input_starcraft.lua#L845